lib/reconcile/cassandra.go

```diff
@@ -286,6 +286,21 @@ func getCassandraPodSpec(statefulSetName, dataVolumeName string, cr *apiv1alpha1
 		},
 	}
 
+	// do we want priorities?
+	if cr.Spec.Features.AstartePodPriorities.IsEnabled() {
+		//is a priorityClass specified in the Astarte CR?
+		switch cr.Spec.Cassandra.AstarteGenericClusteredResource.PriorityClass {
+		case highPriority:
+			ps.PriorityClassName = AstarteHighPriorityName
+		case midPriority:
+			ps.PriorityClassName = AstarteMidPriorityName
+		case lowPriority:
+			ps.PriorityClassName = AstarteLowPriorityName
+		default:
+			ps.PriorityClassName = AstarteMidPriorityName
+		}
```

Comment:
This logic is spread across different files. Just wondering if condensing the logic in a helper function is feasible.

Reply:
I fear it would not be nice, as for example Cassandra is not of type `apiv1alpha1.AstarteComponent`. You'll have two nested switch statements, one for the resource and one for the priority class... Doesn't seem much better